libs/assertion/src/assertion.cpp

```diff
@@ -0,0 +1,48 @@
+//  Copyright (c) 2019 Thomas Heller
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/assertion.hpp>
+
+#include <iostream>
+#include <string>
+
+namespace hpx { namespace assertion {
+    namespace {
```

Comment:
What's the rationale of having this in an unnamed namespace (and not in `namespace detail` for instance)?